kornia/x/trainer.py

```diff
@@ -79,7 +79,7 @@ def __init__(
         self.model = self.accelerator.prepare(model)
         self.train_dataloader = self.accelerator.prepare(train_dataloader)
         self.valid_dataloader = self.accelerator.prepare(valid_dataloader)
-        self.criterion = criterion.to(self.device)
+        self.criterion = None if criterion is None else criterion.to(self.device)
```

Comment:
```suggestion
        self.criterion = criterion or criterion.to(self.device)
```

Reply:
Then you need to update typing but I don’t really like this hack down or the Trainer. Better if we could workaround from object detection trainer to leave this clean